cardano-cli/src/Cardano/CLI/Option/Flag.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Cardano.CLI.Option.Flag
+  ( Flag (Flag)
+  , FlagOptions (FlagOptions)
+  , Defaultness (..)
+  , isDefault
+  , mkFlag
+  , parserFromFlags
+  )
+where
+
+import Cardano.CLI.Option.Flag.Types
+  ( Defaultness (IsDefault)
+  , Flag (Flag)
+  , FlagOptions (FlagOptions)
+  )
+import Cardano.CLI.Option.Flag.Types qualified as Z
+import Cardano.CLI.Vary
+import Cardano.CLI.Vary qualified as Vary
+
+import Control.Applicative
+import Data.Function
+import Options.Applicative (Parser)
+import Options.Applicative qualified as Opt
+
+-- | Create a parser from a help rendering function and list of flags.
+-- A default parser is included at the end of parser alternatives for
+-- the default flag (there should only be one default, but if more than
+-- one is specified, the first such one is used as the default).
+parserFromFlags :: (Flag a -> String) -> [Flag a] -> Parser a
+parserFromFlags _ [] = empty
+parserFromFlags mkHelp fs =
+  alternatives fs <|> defaults fs
+ where
+  alternatives [] = empty
+  alternatives (x : xs) =
+    parserFromFlag mkHelp x <|> alternatives xs
+
+  defaults :: [Flag a] -> Parser a
+  defaults [] = empty
+  defaults (x : xs) =
+    parserFromFlagDefault x <|> defaults xs
+
+parserFromFlag :: (Flag a -> String) -> Flag a -> Parser a
+parserFromFlag mkHelp flag =
+  Opt.flag' (flag & Z.value) $
+    mconcat
+      [ Opt.long $ flag & Z.longName
+      , Opt.help $ mkHelp flag
+      ]
+
+parserFromFlagDefault :: Flag a -> Parser a
+parserFromFlagDefault flag =
+  pure $ flag & Z.value
+
+mkFlag
+  :: a :| fs
+  => String
+  -> String
+  -> a
+  -> Flag (Vary fs)
+mkFlag longName format value =
+  Flag longName format mempty (Vary.from value)
+
+isDefault :: Flag a -> Flag a
+isDefault ff =
```

Comment:
`is` prefix implies that the function does not do any modification and returns Bool. 
```suggestion
setDefault :: Flag a -> Flag a
setDefault ff =
```